### ISSUE TITLE
jsondoclint: False positive for `#![feature(trait_alias)]`

### ISSUE THREAD

**Issue by aDotInTheVoid** [rust-lang/rust#104923]
Very much in the same vane as #104851 (trying to get jsondoclint to work on `core.json`)

```rust
#![feature(trait_alias)]

pub trait Orig<T> {}

pub trait Alias = Orig<i32>;

pub struct Struct;

impl Orig<i32> for Struct {}

pub fn takes_alias(_: impl Alias) {}

fn main() {
    takes_alias(Struct);
}
```

```
---- [rustdoc-json] src/test/rustdoc-json/traits/trait_alias.rs stdout ----

error: jsondoclint failed!
status: exit status: 1
command: "/home/nixon/dev/rust/rust/build/x86_64-unknown-linux-gnu/stage0-tools-bin/jsondoclint" "/home/nixon/dev/rust/rust/build/x86_64-unknown-linux-gnu/test/rustdoc-json/traits/trait_alias/trait_alias.json"
stdout: none
--- stderr -------------------------------
0:5:1566: Expected Trait but found TraitAlias (found at ["$.index[\"0:0:1405\"].inner.items[1]", "$.index[\"0:5:1566\"].id", "$.index[\"0:9:1568\"].inner.decl.inputs[0][1].inner[0].trait_bound.trait.id", "$.index[\"0:9:1568\"].inner.generics.params[0].kind.type.bounds[0].trait_bound.trait.id"])
0:5:1566: Expected Trait but found TraitAlias (found at ["$.index[\"0:0:1405\"].inner.items[1]", "$.index[\"0:5:1566\"].id", "$.index[\"0:9:1568\"].inner.decl.inputs[0][1].inner[0].trait_bound.trait.id", "$.index[\"0:9:1568\"].inner.generics.params[0].kind.type.bounds[0].trait_bound.trait.id"])
Error: Errors validating json /home/nixon/dev/rust/rust/build/x86_64-unknown-linux-gnu/test/rustdoc-json/traits/trait_alias/trait_alias.json
------------------------------------------
```

(This isn't fully standard jsondoclint output, as I've hacked it up to debug this)

@rustbot modify labels: +A-rustdoc-json +A-testsuite +T-rustdoc

@rustbot claim
